compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -485,35 +490,35 @@ pub(crate) fn global_llvm_features(sess: &Session, diagnostics: bool) -> Vec<Str
                 }
                 diag.emit();
             }
-            Some((enable_disable, feature))
+
+            if diagnostics && tied_features.iter().any(|tied| tied.contains(&feature)) {
```

Comment:
This is doing a second linear lookup through the `tied_features` list (in addition to the one that will happen later in the `check_tied_features`.

Ignoring entirely the fact that what's going on here is an exercise in micro-optimization, and the overall performance of the compiler is likely going to stay pretty much the same, I feel like such double lookup is probably defeating any improvements gained here by all the other changes.

It would seem to me that we'd be better off by just not checking this here (while leaving everything else intact):

```suggestion
            if diagnostics {
```

Reply:
Thanks for pointing out! That makes sense and I think we can leave the FIXME here